### PR TITLE
Add an explanatory error message if there is no eth provider available

### LINKED ISSUE #305: [JSTC-26] Add "required" RPC URL & improve error logging
## Summary

When using the CLI, any on-chain action needs an RPC URL. However, if one is not specified, the error logged is ambiguous. The `init` command also notes this URL is optional, but on-chain things won't work unless one is specified.

Note: This isn't *really* a bug, but it sort of is if based on what a user would expect here.

## Details

Set up the CLI for the first time with `tableland init` using `maticmum` and no specified RPC URL. Then, try to a create a table with `tableland create`. You'll get the following error:

```
unsupported getDefaultProvider network (operation="getDefaultProvider", network={
  "chainName": "maticmum",
  "chainId": 80001,
  "contractAddress": "0x4b48841d4b32C4650E4ABc117A03FE8B51f38F68",
  "baseUrl": "https://testnets.tableland.network/api/v1",
  "name": "maticmum"
}, code=NETWORK_ERROR, version=providers/5.7.2)
```

A few ideas:

* Use the ethers default RPC URL if no URL is specified—I *thought* this is how the CLI worked, but based on the error, it doesn't look like it?
* Depending on the bullet above, the `init` command should either say `recommended` or `required` for the `Enter a default JSON RPC API provider URL` question.
* Also depending on the first bullet, if there's no default provider, we should require specifying a provider URL.
* Change the logging to catch the error above and state something like "No provider URL specified"

<sub>From [SyncLinear.com](https://synclinear.com) | [JSTC-26](https://linear.app/tableland/issue/JSTC-26/add-required-rpc-url-and-improve-error-logging)</sub>

### PR DESCRIPTION
**Pull request by joewagner**
Fixes #305

### CODE REVIEW

**Review by joewagner — COMMENTED** [review]
This is RFR

File: src/utils.ts [hunk 1]
```diff
@@ -77,8 +77,15 @@ export async function getWalletWithProvider({
...
   if (!provider) {
-    // This will be significantly rate limited, but we only need to run it once
-    provider = getDefaultProvider({ ...network, name: network.chainName });
+    try {
+      // This will be significantly rate limited, but we only need to run it once
+      provider = getDefaultProvider({ ...network, name: network.chainName });
+    } catch (err: any) {
+      // ethers.js only gives away default provider keys for some networks
+      throw new Error(
+        "no default provider is available for this network, you must provide one via flag (`-p` or `--providerUrl`)"
```
[the remark this answers is not in the document]
Comment (carsonfarmer) [reply src/utils.ts:86]: Yeh nice. I wonder if we should just have a "mapping" of error messages at some global spot in the package to make this kind of thing easier to update/catch?

[the remark this answers is not in the document]
Comment (carsonfarmer) [reply src/utils.ts:86]: Not worth doing for this PR, but something to consider?

**Review by carsonfarmer — APPROVED** [review]

File: src/utils.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (joewagner) [reply src/utils.ts:86]: That's a great idea!  Maybe do a PR to refactor/cleanup this kind of thing everywhere?

**Pull request merged**